Build failed: arithm_core.hpp: No such file or directory

```
[ 31%] Generating precomp.hpp.gch/opencv_core_DEBUG.gch
/[…]/modules/core/precomp.hpp:89:27: fatal error: arithm_core.hpp: No such file or directory
compilation terminated.
modules/core/CmakeFiles/pch_Generate_opencv_core.dir/build.make:62: recipe for target 'modules/core/precomp.hpp.gch/opencv_core_DEBUG.ghc' failed
```

##### System information (version)
- OpenCV => 3.4.1-dev
- Operating System / Platform => Ubuntu 16.04, amd64
- gcc 5.4.0

##### Detailed description
Commit 9e0dee12598273c974c482643b1bb9436e879d7a and later exhibit this issue.
Commit 6e13ffe162b1e6ec79bcbea0dd609f33aef23051 and earlier build fine.

##### Steps to reproduce

<!-- to add code example fence it with triple backticks and optional file extension
    ```.cpp
    // C++ code example
    ```
 or attach as .txt or .zip file
-->

--- a/cmake/OpenCVModule.cmake
+++ b/cmake/OpenCVModule.cmake
@@ -698,6 +698,9 @@ endmacro()
 # setup include path for OpenCV headers for specified module
 # ocv_module_include_directories(<extra include directories/extra include modules>)
 macro(ocv_module_include_directories)
+  if(ENABLE_PRECOMPILED_HEADERS OR OPENCV_INCLUDE_DIR_APPEND_MODULE_SRC)
+    ocv_target_include_directories(${the_module} "${OPENCV_MODULE_${the_module}_LOCATION}/src")
+  endif()
   ocv_target_include_directories(${the_module}
       "${OPENCV_MODULE_${the_module}_LOCATION}/include"
       "${CMAKE_CURRENT_BINARY_DIR}" # for precompiled headers

